src/new.rs

```diff
@@ -527,11 +525,31 @@ pub fn hello_world_test() {{
 }
 
 pub fn create(options: NewOptions, version: &'static str) -> Result<()> {
-    validate_name(&options.name)?;
-    validate_root_folder(&options.name)?;
-    let creator = Creator::new(options, version);
+    let name: String = if options.name.is_empty() {
+        get_foldername(&options.project_root).gleam_expect("unable to get folder name")
+    } else {
+        options.name
+    };
+    validate_name(&name)?;
+    validate_root_folder(&name)?;
+    let creator = Creator::new(
+        NewOptions {
+            name,
+            description: options.description.clone(),
+            project_root: options.project_root.clone(),
+            template: options.template,
+        },
+        version,
+    );
```

Comment:
I think we could mutate the options to avoid creating a new one.

```suggestion
    options.name = name;
    let creator = Creator::new(options, version);
```

We will need to mark the options argument as `mut`.